typescript/lib/config.ts

```diff
@@ -208,8 +208,11 @@ export class AIConfigRuntime implements AIConfig {
    * @param saveOptions Options that determine how to save the AIConfig to the file.
    */
   public save(filePath?: string, saveOptions?: SaveOptions) {
+    const keysToOmit = ['filePath'] as const;
+  
     try {
-      let aiConfigObj: AIConfigRuntime = _.cloneDeep(this);
+      // Create a Deep Copy and omit fields that should not be saved
+      let aiConfigObj: Omit<AIConfigRuntime, typeof keysToOmit[number]> = _.omit(_.cloneDeep(this), keysToOmit)
```

Comment:
Can use `const` here since it's never reassigned

Reply:
updated